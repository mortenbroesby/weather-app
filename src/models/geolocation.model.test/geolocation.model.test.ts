import { GeolocationModel } from "../geolocation.model";

describe("models/geolocation.model.ts", () => {
  it("Generates a valid Geolocation model", () => {
    const SampleJson = require("./geolocation.example.valid");
    const model = new GeolocationModel(SampleJson);

    expect(model).toMatchSnapshot();
  });

  it("Generates a valid Geolocation model with poor data", () => {
    const SampleJson = require("./geolocation.example.poorData");
    const model = new GeolocationModel(SampleJson);

    expect(model).toMatchSnapshot();
  });

  it("Generates a valid Geolocation model with malformed data", () => {
    const SampleJson = require("./geolocation.example.malformed");
    const model = new GeolocationModel(SampleJson);

    expect(model).toMatchSnapshot();
  });

  it("Generates a valid Geolocation model on malformed data", () => {
    const model = new GeolocationModel();

    expect(model).toMatchSnapshot();
  });
});
