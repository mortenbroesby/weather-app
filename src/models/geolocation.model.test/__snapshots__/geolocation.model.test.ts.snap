// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/geolocation.model.ts Generates a valid Geolocation model 1`] = `
GeolocationModel {
  "location": Object {
    "coords": Object {
      "latitude": 52.3846,
      "longitude": 4.9172,
    },
  },
}
`;

exports[`models/geolocation.model.ts Generates a valid Geolocation model on malformed data 1`] = `
GeolocationModel {
  "location": Object {
    "coords": Object {
      "latitude": 52.3886,
      "longitude": 4.9068,
    },
  },
}
`;

exports[`models/geolocation.model.ts Generates a valid Geolocation model with malformed data 1`] = `
GeolocationModel {
  "location": Object {
    "coords": Object {
      "latitude": 52.3886,
      "longitude": 4.9068,
    },
  },
}
`;

exports[`models/geolocation.model.ts Generates a valid Geolocation model with poor data 1`] = `
GeolocationModel {
  "location": Object {
    "coords": Object {
      "latitude": 52.3886,
      "longitude": 4.9068,
    },
  },
}
`;
